python/ql/src/experimental/Security/CWE-091/XsltConcept.qll

```diff
@@ -0,0 +1,110 @@
+private import python
+private import semmle.python.dataflow.new.DataFlow
+private import semmle.python.ApiGraphs
+
+/**
+ * A data-flow node that constructs a XSLT transformer.
+ *
+ * Extend this class to refine existing API models. If you want to model new APIs,
+ * extend `TemplateConstruction::Range` instead.
+ */
+class XsltConstruction extends DataFlow::Node instanceof XsltConstruction::Range {
+  /** Gets the argument that specifies the XSLT transformer. */
+  DataFlow::Node getXsltArg() { result = super.getXsltArg() }
+}
+
+/** Provides a class for modeling new system-command execution APIs. */
+module XsltConstruction {
+  /**
+   * A data-flow node that constructs a XSLT transformer.
+   *
+   * Extend this class to model new APIs. If you want to refine existing API models,
+   * extend `XsltConstruction` instead.
+   */
+  abstract class Range extends DataFlow::Node {
+    /** Gets the argument that specifies the XSLT transformer. */
+    abstract DataFlow::Node getXsltArg();
+  }
+}
+
+/**
+ * A data-flow node that executes a XSLT transformer.
+ *
+ * Extend this class to refine existing API models. If you want to model new APIs,
+ * extend `TemplateConstruction::Range` instead.
+ */
+class XsltExecution extends DataFlow::Node instanceof XsltExecution::Range {
+  /** Gets the argument that specifies the XSLT transformer. */
+  DataFlow::Node getXsltArg() { result = super.getXsltArg() }
+}
+
+/** Provides a class for modeling new system-command execution APIs. */
+module XsltExecution {
+  /**
+   * A data-flow node that executes a XSLT transformer.
+   *
+   * Extend this class to model new APIs. If you want to refine existing API models,
+   * extend `XsltExecution` instead.
+   */
+  abstract class Range extends DataFlow::Node {
+    /** Gets the argument that specifies the XSLT transformer. */
+    abstract DataFlow::Node getXsltArg();
+  }
+}
+
+// -----------------------------------------------------------------------------
```

Comment:
From here seems to be converted modelling from `experimental.semmle.python.security.injection.XSLT`.

Reply:
yep :+1: I've removed the `XSLT.qll` file now :+1: (
0dca8a5)